# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1386365996 18000
#      Fri Dec 06 16:39:56 2013 -0500
# Branch stable
# Node ID 0fa235a252a5c26ea2f539469742bc7152de4101
# Parent  a6cfe292bfb6b5c664071abc76879aa941a0497d
fix cut and paste error in Matrix by ComplexMatrix multiplication operator

* fCMatrix.cc (operator * (const FloatMatrix&, const
FloatComplexMatrix&)): Fix cut and paste error.

diff --git a/liboctave/array/fCMatrix.cc b/liboctave/array/fCMatrix.cc
--- a/liboctave/array/fCMatrix.cc
+++ b/liboctave/array/fCMatrix.cc
@@ -3750,17 +3750,17 @@ operator * (const FloatComplexMatrix& m,
 }
 
 FloatComplexMatrix
 operator * (const FloatMatrix& m, const FloatComplexMatrix& a)
 {
   if (a.rows () > std::min (m.rows (), a.columns ()) / 10)
     return FloatComplexMatrix (m * real (a), m * imag (a));
   else
-    return m * FloatComplexMatrix (a);
+    return FloatComplexMatrix (m) * a;
 }
 
 /*
 
 ## Simple Dot Product, Matrix-Vector, and Matrix-Matrix Unit tests
 %!assert (single ([1+i 2+i 3+i]) * single ([ 4+i ; 5+i ; 6+i]), single (29+21i), 5e-7)
 %!assert (single ([1+i 2+i ; 3+i 4+i]) * single ([5+i ; 6+i]), single ([15 + 14i ; 37 + 18i]), 5e-7)
 %!assert (single ([1+i 2+i ; 3+i 4+i ]) * single ([5+i 6+i ; 7+i 8+i]), single ([17 + 15i 20 + 17i; 41 + 19i 48 + 21i]), 5e-7)
